client/src/components/ShareFundDialog.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from "react";
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  DialogActions,
+  Button,
+  InputLabel,
+  Select,
+  MenuItem,
+  FormControl
+} from "@material-ui/core";
+import FileInput from "./FileInput";
+import axios from "axios";
+import { User } from "../LoginPage";
+import Axios from "axios";
+
+export interface ShareFundDialogProps {
+  fundId: string;
+  handleClose: Function;
+}
+
+export interface ShareFundDialogState {
+  open: boolean;
+  users: User[];
+  selectedUser?: User;
+}
+
+export default class ShareFundDialog extends React.Component<ShareFundDialogProps, ShareFundDialogState> {
+  state: ShareFundDialogState = { open: true, users: [] };
+
+  componentDidMount() {
+    this.getUsers().then((users: User[]) => {
+      this.setState({ users });
+    });
+  }
+
+  getUsers = async () => {
+    return new Promise<User[]>((resolve, reject) => {
+      Axios.get("/users")
+        .then(response => resolve(response.data.users))
+        .catch(function(error) {
+          console.log(error);
+          reject();
+        });
+    });
+  };
+
+  handleUserSelected = (event: React.ChangeEvent<{ value: unknown }>) => {
+    this.setState({ selectedUser: this.state.users[event.target.value as number] });
+  };
+
+  handleShare = () => {
+    if (this.state.selectedUser) {
+      this.shareFundWithUser(this.props.fundId, this.state.selectedUser).finally(() => {
+        this.props.handleClose();
+      });
+    }
+  };
+
+  shareFundWithUser = async (fundId: string, user: User) => {
+    return new Promise<User[]>((resolve, reject) => {
+      Axios.post("/shareFund", { fundId, user })
```

Comment:
Axios is the tool used to have the front end contact the backend. Server.ts has API endpoints using app.get("/FunctionName").